modules/standard/CTypes.chpl

```diff
@@ -461,24 +548,43 @@ module CTypes {
   pragma "no doc"
   inline operator c_ptr.-(a: c_ptr, b: integral) do return __primitive("-", a, b);
 
+  pragma "no doc"
+  inline operator c_ptrConst.!(x: c_ptrConst) do return x == c_nil;
+
+  pragma "no doc"
+  inline operator c_ptrConst.+(a: c_ptrConst, b: integral) do
+    return __primitive("+", a, b);
+
+  pragma "no doc"
+  inline operator c_ptrConst.-(a: c_ptrConst, b: integral) do
+    return __primitive("-", a, b);
+
   pragma "no doc"
   inline operator c_ptr.-(a: c_ptr(?t), b: c_ptr(t)):c_ptrdiff {
     return c_pointer_diff(a, b, c_sizeof(a.eltType):c_ptrdiff);
   }
 
+  pragma "no doc"
+  inline operator c_ptrConst.-(a: c_ptrConst(?t), b: c_ptrConst(t)):c_ptrdiff {
+    return c_pointer_diff(a, b, c_sizeof(a.eltType):c_ptrdiff);
+  }
+
   pragma "no doc"
   pragma "fn synchronization free"
   pragma "codegen for CPU and GPU"
   extern proc c_pointer_return(ref x:?t):c_ptr(t);
   pragma "no doc"
   pragma "fn synchronization free"
+  pragma "codegen for CPU and GPU"
+  extern proc c_pointer_return_const(const ref x:?t):c_ptrConst(t);
```

Comment:
I am pretty sure this calls for a sibling of this here: https://github.com/chapel-lang/chapel/blob/main/runtime/include/gpu/cuda/chpl-gpu-gen-includes.h#L62 and also in a similar header for rocm layer. I can help you with testing it, but maybe it'll be easier for others if you ask for it next week.

Reply:
Oh, thanks for calling that out, I had no idea it existed. I will leave it to next week since I don't wanna break any GPU stuff and don't know what I'm doing there.

Reply:
Per discussion in GPU thread, I am not able to do the GPU tests as they are currently failing on main, but will merge anyways as this is unlikely to break. I added a test exercising the GPU runtime changes.